### ISSUE TITLE
baseline performance, will be improved?

### ISSUE THREAD

**Issue by rquintino** [Azure/doAzureParallel#34]
Hi, I was trying this with a cluster of low priority standard_d5_v2 vms, 3 nodes, 16 max tasks per node

**minimum time I get is around 20secs for the bare minimum task-just outputting hostname, something I'm missing?**
or will it be improved?

**_code_**
number_of_iterations=10
system.time(results <- foreach(i = 1:number_of_iterations) %dopar% {
  (system('hostname',intern=T))
})
results

**_output_**
[1] "Job Summary: "
[1] "Id: job20170629182126"
[1] "Waiting for tasks to complete. . ."
  |=========================================================| 100%[1] "Number of errors: 0"
   user  system elapsed 
   **0.86    0.09   22.22** 
There were 14 warnings (use warnings() to see them)
> results
[[1]]
[1] "batchnode-0"

[[2]]
[1] "batchnode-2"

[[3]]
[1] "batchnode-1"

[[4]]
[1] "batchnode-0"

[[5]]
[1] "batchnode-1"

[[6]]
[1] "batchnode-2"

[[7]]
[1] "batchnode-0"

[[8]]
[1] "batchnode-0"

[[9]]
[1] "batchnode-0"

[[10]]
[1] "batchnode-2"

**Comment by rquintino** [comment]
just for comparison, using a doRedis/docker swarm cluster local 7 lowend nodes, 3 workers each, ~21cores I get <1 sec response.

know doazureparallel is targeted at long jobs but still, 20 secs seems to much

user  system elapsed 
   **_0.07    0.00    0.94_** 
> results
[[1]]
[1] "dc6918d27f75"

[[2]]
[1] "d7e1a7e38d73"

[[3]]
[1] "8ffb2fb13cf6"

[[4]]
[1] "ab0127b9410a"

[[5]]
[1] "2d8a79926519"

[[6]]
[1] "99d556ea5e02"

[[7]]
[1] "31a08d3b17b7"

[[8]]
[1] "cc85b9884f43"

[[9]]
[1] "a2eea050ec98"

[[10]]
[1] "2a0280025f3d"

**Comment by paselem** [comment]
The 20 seconds you're seeing is due to how we poll for task states. We do target longer running jobs, and polling can add a lot of network traffic on your machine.

That said, we could technically expose a way to customize the poll time, but that may be overkill. In the short-term It's probably worth just submitting a job with wait=False and polling yourself if you expect the job will run very quickly.

Note that this may be blocked on #32 at the moment.

**Comment by paselem** [comment]
@rquintino - we just released our long running job support which allows you to poll for job completion at your own rate (not limited to our polling time). I recommend taking a look at this although it will still not be nearly as fast as your local cluster for such small workloads, but that is not the kind of workload we target. There is a break-even point for at least a few minutes of runtime until we meet or exceed the performance of a local cluster. For small jobs like the one you have shown above, we recommend continuing to use use a local cluster (or doParallel's ability to leverage the cores on your local machine). For larger/longer-running workloads we feel doAzureParallel will have big benefits.

Thanks,
-Pablo

**Comment by rquintino** [comment]
Hi Pablo, thanks for the tip, 
finding the long running and non blocking job support really useful, was using futures for non blocking but this is easier & requires fewer resources. 
btw-Plans for job cancellation support? :)

**Comment by paselem** [comment]
Hrm.. Seems like an oversight on our part to not have those exposed. Those functions exist in rAzureBatch though, so you should be able to do something like rAzureBatch::terminateJob(job_id) and rAzureBatch::deleteJob(job_id) as a work around.

Note: If you call deleteJob directly, it will automatically terminate it.
